### PR TITLE
feat: add trl template

### PR DESCRIPTION
**Pull request by phala-agent**
## Summary
- Add a `trl` prebuilt template for `huggingface/trl`.
- Runs a CPU-safe HTTP verifier that installs/imports the real TRL package and checks trainer/config/data-flow metadata.
- Exposes `/healthz`, `/demo`, and `/v1/models` without model downloads, GPU/CUDA, provider credentials, or real training by default.

## Template files
- `templates/prebuilt/trl/docker-compose.yml`
- `templates/prebuilt/trl/README.md`
- `templates/icons/trl.png`
- `templates/config.json`

## Configuration / env vars
- `TRL_PACKAGE_VERSION` optional, default `0.24.0`
- `PYTORCH_CPU_VERSION` optional, default `2.9.1+cpu`
- `TRANSFORMERS_PACKAGE_VERSION` optional, default `4.57.6`
- `ACCELERATE_PACKAGE_VERSION` optional, default `1.13.0`
- `DATASETS_PACKAGE_VERSION` optional, default `4.8.5`

No secrets are required by the default verifier.

## Icon source
- Upstream `huggingface/trl` asset `assets/logo-dark.png`, stored as `templates/icons/trl.png`.

## Validation
- `python3 templates/validate.py` ✅
- `git diff --check origin/main...HEAD` ✅
- `docker compose -f templates/prebuilt/trl/docker-compose.yml config >/dev/null` ✅
- Static audit ✅: README coverage, config ID uniqueness, icon existence, no literal secrets, no `env_file`, no host bind mounts, no privileged/host namespace, no external build context.

## Phala Cloud smoke test
- Profile/workspace: `hermes-admin-cvm-check` / `h4x's projects`
- Deploy command shape: `phala deploy --profile hermes-admin-cvm-check --name auto-template-test-trl-0527-1518 --compose templates/prebuilt/trl/docker-compose.yml --instance-type tdx.small --disk-size 40G --wait --no-public-logs --no-public-sysinfo --json`
- CVM UUID: `e647a47b-5f54-41dd-ab38-682c8170bb85`
- App ID: `753d323b9d23dfb8efff85561af65dc7c94e8fc0`
- `phala ps` ✅: `/dstack-app-1` running, `Up ... (healthy)`
- Endpoint: `https://753d323b9d23dfb8efff85561af65dc7c94e8fc0-8080.dstack-pha-prod7.phala.network`
- `GET /healthz` ✅ HTTP 200, `trl.import_ok=true`, `torch=2.9.1+cpu`, `cuda_available=false`
- `GET /demo` ✅ HTTP 200, `model_downloaded=false`, `training_started=false`, `trainer_constructed=false`
- `GET /v1/models` ✅ HTTP 200, includes `trl-cpu-safe-verifier`

## Cleanup
- `phala cvms delete --profile hermes-admin-cvm-check e647a47b-5f54-41dd-ab38-682c8170bb85 --force --json` ✅
- Follow-up lookup returned `The requested CVM was not found` ✅

cc @Marvin-Cypher for review.

### CODE REVIEW

**Comment by phala-agent** [comment]
cc @Marvin-Cypher for review.

**Review by Marvin-Cypher — APPROVED** [review]
Reviewed template files, normalized tags, compose config, config validation, stale path cleanup, and compose safety scan. Checks are passing.
